src/gui/folderman.cpp

```diff
@@ -76,6 +76,9 @@ FolderMan::FolderMan(QObject *parent)
     connect(AccountManager::instance(), &AccountManager::removeAccountFolders,
         this, &FolderMan::slotRemoveFoldersForAccount);
 
+    connect(AccountManager::instance(), &AccountManager::accountRemoved,
```

Comment:
Hm I think this is emitted at application exit too... if I'm correct you don't want to call `vfs->unregisterFolder()` there but only `vfs->stop()`. Otherwise IIRC that would kill all the stored placeholder information in the folder every time you exit.

I think what you're after is about unregistering folders for accounts the user deleted? If yes how is that related to what this PR states? Just trying to connect the dots here. :-)

Reply:
> Hm I think this is emitted at application exit too... if I'm correct you don't want to call `vfs->unregisterFolder()` there but only `vfs->stop()`. Otherwise IIRC that would kill all the stored placeholder information in the folder every time you exit.
> 
> I think what you're after is about unregistering folders for accounts the user deleted? If yes how is that related to what this PR states? Just trying to connect the dots here. :-)

@er-vin No. Absolutely no. I've been debugging it and it is not emitted. Only emitted when the account is removed.

Another question - when we are removing the account connection - why would we need to store placeholder information? My idea was  - they are now invalid and we will instead register everything again when adding a new account connection (which actually happens now).

Reply:
> @er-vin No. Absolutely no. I've been debugging it and it is not emitted. Only emitted when the account is removed.

This is surprising, I wonder if something else is broken because this is clearly emitted from `AccountManager::shutdown()` which is called from the `Application` dtor.

> Another question - when we are removing the account connection - why would we need to store placeholder information? My idea was - they are now invalid and we will instead register everything again when adding a new account connection (which actually happens now).

Sure, when the user removes an account there's no need to keep placeholder information. My comment was more from the prospect of my first comment that I expect the folder removal code to be triggered on application exit which we wouldn't want. Placeholder information needs to stay put even if the application is exited.

Reply:
@er-vin Indeed, emit signal is there. But, the `_folderMap` is empty by the time it gets emitted, that's why I am not hitting it. Anyways, thx for this comment, I definitely need to change this part to make it more reliable. Now, it works, just because `_folderMap` is cleaned up in advance, but, it will break if something changes in the cleanup logic for `_folderMap`.